service/dispatcher.js

```diff
@@ -132,10 +138,18 @@ class UserContext extends events.EventEmitter {
         if (this._user.access_token)
             headers['Authorization'] = `Bearer ${this._user.access_token}`;
 
+        let tempToken = this._user.access_token;
+        headers['Authorization'] = `Bearer ` + tempToken;
```

Comment:
This line duplicates the one above?

Reply:
After the WebSocket receives a message from Slack, the connection is closed (after sending a message to Almond). After a minute, the UserContext turns inactive, hence calling the close function (lines 80-87). This close function attempts to close the WebSocket again, but the WebSocket has already been set to null after receiving the message. It is this issue that causes a nonexisting WebSocket to attempt to close, resulting in the error seen. 

As for the style issues, I have incorporated the changes requested.

Reply:
Do you know why the connection gets closed? Server side nothing should close the connection...

Reply:
I'm unsure why this happens. If I had to take a guess, I'd say it's because something is calling the close method on the WebSocket, but I can't pinpoint exactly how or why this is happening. Do you know where I could look to figure this out?

Reply:
Well, the first step is to identify who initiates closing the connection: the client or the server? You control the client code, so you can audit all paths and see if `.close()` is called client-side...

Reply:
Within the dispatcher, the only time the close method is called is when the UserContext turns inactive (after 1 minute). Does this mean that the error occurs server-side?

`        ctx.on('inactive', () => {
            ctx.close();
            this._userContexts.delete(key);
        });
`

Reply:
Yeah something like that. Perhaps it is closed by some middlebox, like our load balancer or your NAT (home router). Because nothing in the server should close the connection except in case of errors or crashes...

Reply:
Oh okay. Do you know what can be done to pinpoint where the issue lies?